modules/data-loading/pages/kafka-loader-user-guide.adoc

```diff
@@ -31,75 +118,118 @@ There are three basic steps:
 
-* If you have a TigerGraph cluster, the configuration file must be on machine m1, where the GSQL server and GSQL client both reside,  and it must be in JSON format. If the configuration file uses a relative path, the path should be relative to the GSQL client working directory.
-* Each time when the config file is updated, you must run "SET data_source_name"  to update the data source details in the dictionary.
-====
+[source,text]
+----
+DEFINE FILENAME f1 = "$k1:~/topic_partition_config.json";
+DEFINE FILENAME f1 = "$k1:{\"topic\":\"zzz\",\"default_start_offset\":2,\"partition_list\":[]}";
+----
 
-To further simplify, instead of specifying the Kafka data source config file path, you can also directly provide the Kafka data source configuration as a string argument, as shown below:
+[#_3_run_the_loading_job]
+=== 3. Run the loading Job
+
+The Kafka Loader uses the same xref:gsql-ref:ddl-and-loading:running-a-loading-job.adoc[`RUN LOADING JOB`] statement that is used for GSQL loading from files.
+Each filename variable can be assigned a string `<data_source_name>:<topic_partition_configuration_filepath>`, which will override the value defined in the loading job.
+
+In the example below, the config files for f3 and f4 are being set by the `RUN command`, whereas f1 is using the config which was specified in the `CREATE LOADING JOB` statement.
 
 [source,gsql]
 ----
-GSQL > CREATE DATA_SOURCE KAFKA data_source_name = "{\"broker\":\"broker.full.domain.name:9092\"}"
+RUN LOADING JOB job1 USING f1, f3="$k1:~/topic_part3_config.json", f4="$k1:~/topic_part4_config.json", EOF="true";
 ----
 
-[NOTE]
+[CAUTION]
 ====
-*Tip*: The above simplified statement is useful for using Kafka Data Loader in TigerGraph Cloud. In TigerGraph Cloud (tgcloud.io), you can use GSQL web shell to define and create Kafka data sources, without creating the Kafka data source configuration file in filesystem.
+A `RUN LOADING JOB` command may only use one type of data source.
+
+For example, you may not mix both Kafka data sources and regular file data sources in one loading job.
 ====
 
-*Local data source vs global data source:*
+All filename variables in one loading job statement must refer to the same `DATA_SOURCE` variable.
+
+There are two modes for the Kafka Loader: streaming mode and EOF mode. The default mode is streaming mode.
+In streaming mode, loading will never stop until the job is aborted. In EOF mode,  loading will stop after consuming the current Kafka message.
+
+To set EOF mode, an optional parameter is added to the `RUN LOADING JOB` syntax:
+
+[source,ebnf]
+----
+RUN LOADING JOB [-noprint] [-dryrun] [-n [i],j] jobname
+   [ USING filevar [="filepath_string"][, filevar [="filepath_string"]]*
+   [, CONCURRENCY="cnum"][,BATCH_SIZE="bnum"]][, EOF="true"]
+----
+
+To learn about each option and parameter of the `RUN LOADING JOB` command, see xref:gsql-ref:ddl-and-loading:running-a-loading-job.adoc#_options[Loading job options].
+
+== Manage data sources
```

Comment:
Consistency of verb tense in headings:

1. There is a heading "Configuring and Using the Kafka Loader" (gerund, analogous to present continuous)
2. This is followed by "Manage data sources" and "Manage loading jobs" (imperative)
3. Also, the subsections under "Configuring and Using the Kafka Loader" use the imperative tense.